pkgs/unix_api/src/libc_shim.c

```diff
@@ -148,3 +150,152 @@ int libc_shim_fstatat(int fd, char *path, struct libc_shim_Stat *buf,
   }
   return r;
 }
+
+// <pthread.h>
+
+int libc_shim_pthread_cond_broadcast(libc_shim_pthread_cond_t *cond, int *err) {
+  errno = *err;
+  int r = pthread_cond_broadcast((pthread_cond_t *)cond->_pthread_cond_t);
+  *err = errno;
+  return r;
+}
+
+int libc_shim_pthread_cond_destroy(libc_shim_pthread_cond_t *cond, int *err) {
+  errno = *err;
+  int r = pthread_cond_destroy((pthread_cond_t *)cond->_pthread_cond_t);
+  *err = errno;
+  free(cond->_pthread_cond_t);
+  return r;
+}
+
+int libc_shim_pthread_cond_init(libc_shim_pthread_cond_t *cond,
+                                const libc_shim_pthread_condattr_t *attr,
+                                int *err) {
+  pthread_cond_t *c = (pthread_cond_t *)calloc(1, sizeof(pthread_cond_t));
+  errno = *err;
+  int r = pthread_cond_init(
+      c,
+      (attr == NULL) ? NULL : (pthread_condattr_t *)attr->_pthread_condattr_t);
+  *err = errno;
+  if (r == 0) {
+    cond->_pthread_cond_t = c;
```

Comment:
Is there a need to assert that cond has not been already initialized? Calling this function more than once against same cond will result in a memory leak, right?

Reply:
Not sure. I think I just wished that there would be no malloc/free in these methods. But I don't have a proposal on how to avoid them neither.